impeller/renderer/backend/vulkan/tracked_objects_vk.cc

```diff
@@ -39,40 +44,27 @@ bool TrackedObjectsVK::IsValid() const {
 }
 
 void TrackedObjectsVK::Track(std::shared_ptr<SharedObjectVK> object) {
-  if (!object) {
+  if (!object || (!tracked_objects_.empty() &&
+                  object.get() == tracked_objects_.back().get())) {
     return;
   }
-  tracked_objects_.insert(std::move(object));
+  tracked_objects_.emplace_back(std::move(object));
 }
 
 void TrackedObjectsVK::Track(std::shared_ptr<const DeviceBuffer> buffer) {
-  if (!buffer) {
+  if (!buffer || (!tracked_buffers_.empty() &&
```

Comment:
Ah so always check if this was the last buffer to avoid endless increases? smart.

Reply:
Yea, in some of the unit tests I've seen this drop the number of buffers from 45 to 1.  It isn't a logical error to have multiple entries but hopefully there isn't the pathological case of repeatedly adding (1,2,1,2,1,2,1,2,...)

Reply:
no, i think in reality all buffers will either be the same or change over once. So either 1, 1, 1, 1, 1 or 1, 1, 1, 2, 2, 2